### PR TITLE
syncer: fix dm skip mariadb event problem

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/5394

### What is changed and how it works?

Update current location once we met new gtid events.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - None

Side effects

 - Possible performance regression

Related changes

 - None

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Comment by lance6716** [comment]
let's check https://github.com/pingcap/tiflow/actions/runs/2310706543

File: dm/syncer/syncer.go [hunk 1]
```diff
@@ -1853,6 +1854,28 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
...
+				return err2
+			}
+			if lastLocation.GetGTID().Contain(gtidSet) {
+				return nil
+			}
+		}
+
+		// clone lastLocation's gtid set to avoid its gtid is transport to table checkpoint
+		// currently table checkpoint will save last location's gtid set with shallow copy
+		lastLocation = lastLocation.Clone()
```
[the remark this answers is not in the document]
Comment (GMHDBJD) [reply dm/syncer/syncer.go:1870]: Can we use lastLocation.SetGTID?

File: dm/syncer/syncer.go [hunk 2]
```diff
@@ -1853,6 +1854,27 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		return nil
 	}
 
+	advanceLatestLocationGtidSet := func(e *replication.BinlogEvent) error {
+		if _, ok := e.Event.(*replication.MariadbGTIDEvent); ok {
+			gtidSet, err2 := gtid.ParserGTID(s.cfg.Flavor, currentGTID)
```
Comment (GMHDBJD) [inline dm/syncer/syncer.go:1859]: At L2279 we get gtidSet as string, here we parse gtid string to gtidSet, can we improve it?

Comment (lichunzhu) [reply dm/syncer/syncer.go:1859]: https://github.com/go-mysql-org/go-mysql/blob/4b20384/replication/binlogsyncer.go#L787
GTIDSet only accepts updates by string now. We need to update go-mysql-org if we don't want to parse gtid string again. Maybe later.

File: dm/syncer/syncer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (lichunzhu) [reply dm/syncer/syncer.go]: updated in https://github.com/pingcap/tiflow/pull/5401/commits/d03d7e2435e135b564432ac3fe450256c2177670

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by GMHDBJD** [comment]
Double check https://github.com/pingcap/tiflow/actions/runs/2312403672

**Review by lance6716 — APPROVED** [review]

**Comment by lichunzhu** [comment]
/run-dm-integration-test
/run-verify-ci
https://ci2.pingcap.net/job/dm_ghpr_integration_test/5602/display/redirect
https://ci.pingcap.net/job/tiflow-verify-go1.18-pipeline/2861/display/redirect

**Pull request merged**